infra/charts/turing/templates/_helpers.tpl

```diff
@@ -165,6 +165,9 @@ initContainers:
 {{- define "turing.defaultConfig" -}}
 ClusterConfig:
   InClusterConfig: {{ .Values.turing.clusterConfig.useInClusterConfig }}
+  EnvironmentConfigPath: {{ include "turing.environments.absolutePath" . }}
+  EnsemblingServiceK8sConfig: 
+{{ .Values.turing.clusterConfig.ensemblingServiceK8sConfig | toYaml | indent 4}}
```

Comment:
@deadlycoconuts, thanks for concretely noting this concern. I agree with you 100%. But I didn't raise this earlier because I think that, unfortunately the decision to keep 2 separate configs is also driven by Merlin.
1. Merlin is currently using the list of envConfigs for real-time + batch deployments (which needs some improvement too, to classify them better, as we were discussing earlier).
2. Merlin has a separate imagebuilder config for the Pyfunc image building cluster info

Ref: `api/cmd/api/setup.go` in https://github.com/gojek/merlin/pull/331/files

The code changes in Turing are also reflective of that. It fully makes sense to let `envConfigs` have all the configs and the ImageBuilder.K8sConfig (Merlin) / EnsemblingServiceK8sConfig (Turing) only hold the names of the clusters, as you have pointed out. And we can probably make this change right away for Turing. But I think it will be easier to do this for both apps together later on, and also address the other dangling problems with the handling of environments in CaraML (i.e., move the `/environments` API to MLP app).

Does it make sense?

Reply:
Ah I see, I wasn't aware Merlin was managing the clusterconfigs separately too 😯 Yeah that makes sense, we can clean all those up in one go with the migration of the `/environments` API to the MLP app 🙏 

Thanks a lot for the input! Resolving this thread now.

Reply:
Thanks @krithika369 for the clarification, yep i didn't want to make too big a change to the existing flow in 1 PR, in case it might affect other areas that I wasn't aware of.